src/test/java/htsjdk/samtools/cram/structure/SliceTests.java

```diff
@@ -72,238 +75,180 @@ public void testFailsMD5Check() throws IOException {
-            }
-
-            records.add(record);
-        }
-
-        final CompressionHeader header = new CompressionHeaderFactory().build(records, null, true);
-
+    @Test(dataProvider = "forBuildTests")
+    public void testBuild(final List<CramCompressionRecord> records,
+                          final boolean coordinateSorted,
+                          final int expectedRecordCount,
+                          final int expectedSequenceId,
+                          final int expectedAlignmentStart,
+                          final int expectedAlignmentSpan) {
+        final CompressionHeader header = new CompressionHeaderFactory().build(records, null, coordinateSorted);
         final Slice slice = Slice.buildSlice(records, header);
-
-        Assert.assertEquals(slice.nofRecords, 10);
-        Assert.assertEquals(slice.sequenceId, Slice.MULTI_REFERENCE);
-        Assert.assertEquals(slice.alignmentStart, Slice.NO_ALIGNMENT_START);
-        Assert.assertEquals(slice.alignmentSpan, Slice.NO_ALIGNMENT_SPAN);
+        assertSliceState(slice, expectedRecordCount, expectedSequenceId, expectedAlignmentStart, expectedAlignmentSpan);
     }
 
     // show that a slice with a single ref will initially be built as single-ref
-    // but adding an additional refs will make it multiref
-    // and another will keep it multiref
+    // but adding an additional ref will make it multiref
+    // and more will keep it multiref (mapped or otherwise)
 
     @Test
     public void testBuildStates() {
         final List<CramCompressionRecord> records = new ArrayList<>();
 
-        final CramCompressionRecord record0 = new CramCompressionRecord();
-        record0.readBases = "AAA".getBytes();
-        record0.qualityScores = "!!!".getBytes();
-        record0.readLength = 3;
-        record0.readName = "1";
-        record0.sequenceId = 0;
-        record0.alignmentStart = 1;
-        record0.setLastSegment(true);
-        record0.readFeatures = Collections.emptyList();
-        record0.setSegmentUnmapped(false);
-
-        records.add(record0);
-
-        final CompressionHeader header0 = new CompressionHeaderFactory().build(records, null, true);
-        final Slice slice0 = Slice.buildSlice(records, header0);
-
-        Assert.assertEquals(slice0.nofRecords, 1);
-        Assert.assertEquals(slice0.sequenceId, 0);
-        Assert.assertEquals(slice0.alignmentStart, 1);
-        Assert.assertEquals(slice0.alignmentSpan, 3);
-
-        final CramCompressionRecord record1 = new CramCompressionRecord();
-        record1.readBases = "AAA".getBytes();
-        record1.qualityScores = "!!!".getBytes();
-        record1.readLength = 3;
-        record1.readName = "1";
-        record1.sequenceId = 1;
-        record1.alignmentStart = 1;
-        record1.setLastSegment(true);
-        record1.readFeatures = Collections.emptyList();
-        record1.setSegmentUnmapped(false);
-
+        final CramCompressionRecord record1 = createRecord(0, 0);
```

Comment:
It seems like this test has a lot of duplicated code, and is doing something very similar it what `getSingleRefRecords`/`getMultiRefRecords` is doing. It is possible to rewrite it test a provide and those primitives ?